src/logic/calculate.js

```diff
@@ -0,0 +1,76 @@
+import operate from './operate';
+
+const calculate = (data, btnName) => {
+  let { total, next, operation } = data;
+
+  switch (btnName) {
+    case '+/-':
+      if (total) total = operate(total, '-1', 'x');
+      else if (next) next = operate(next, '-1', 'x');
+      break;
+    case '%':
+      if (next) {
+        total = operate(total, next, operation);
+        total = operate(total, '100', '÷');
+        next = null;
+        operation = null;
+      } else {
+        total = operate(total, '100', '÷');
+      }
+      break;
```

Comment:
## Fix `%` operator functionality.
- [ ] During a new operation, when I click on a number and then the `%` operator, `React` throws the following error. 👇 

![calculator-demo](https://user-images.githubusercontent.com/22345961/107078417-5f697400-67ff-11eb-8e77-36a59d9575de.gif)

- However, on the positive though, after an operation, the operation works just fine. Kindly attend to this before we proceed.